helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -622,7 +645,8 @@ void unregisterMessageHandlerFactory(String type) {
             + ", pool: " + pool);
 
     if (pool != null) {
-      shutdownAndAwaitTermination(pool);
+      int timeout = item == null? DEFAULT_MSG_HANDLER_RESET_TIMEOUT_MS : item.getResetTimeout();
+      shutdownAndAwaitTermination(pool, timeout);
```

Comment:
Nit: consider moving line 648 into shutdownAndAwaitTermination to avoid repeating it.

Reply:
Good point. Will change.